File: moderate/point_in_circle.lua
for line in io.lines(arg[1]) do
  _, _, x1, y1, r, x2, y2 = string.find(line, "Center: %((-?%d+.?%d*), (-?%d+.?%d*)%); Radius: (-?%d+.?%d*); Point: %((-?%d+.?%d*), (-?%d+.?%d*)%)")
  local x = x2-x1
  local y = y2-y1
  if x*x + y*y <= r*r then
    print("true")
  else
    print("false")
  end
end

File: moderate/point_in_circle.c
#include <stdbool.h>
#include <stdio.h>

static bool incircle(float a, float b, float c) {
	return a * a + b * b <= c * c;
}

int main(int argc, char *argv[]) {
	FILE *fp;
	float cx, cy, r, px, py;

	fp = fopen(*++argv, "r");
	while (fscanf(fp, "Center: (%f, %f); Radius: %f; Point: (%f, %f)\n", &cx, &cy, &r, &px, &py) != EOF)
		puts(incircle(cx - px, cy - py, r) ? "true" : "false");
	return 0;
}
